test/tests/components/ws-manager/dotfiles_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package wsmanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/rpc"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	csapi "github.com/gitpod-io/gitpod/content-service/api"
+	agent "github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	wsmanapi "github.com/gitpod-io/gitpod/ws-manager/api"
+)
+
+func TestDotfiles(t *testing.T) {
+	userToken, _ := os.LookupEnv("USER_TOKEN")
+	integration.SkipWithoutUsername(t, username)
+	integration.SkipWithoutUserToken(t, userToken)
+
+	f := features.New("dotfiles").WithLabel("component", "ws-manager").Assess("ensure dotfiles are loaded", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
+		api := integration.NewComponentAPI(ctx, cfg.Namespace(), kubeconfig, cfg.Client())
+		t.Cleanup(func() {
+			api.Done(t)
+		})
+
+		userId, err := api.CreateUser(username, userToken)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		tokenId, err := api.CreateOAuth2Token(username, []string{
+			"function:getToken",
+			"function:openPort",
+			"function:getOpenPorts",
+			"function:guessGitTokenScopes",
+			"function:getWorkspace",
+			"resource:token::*::get",
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		swr := func(req *wsmanapi.StartWorkspaceRequest) error {
+			req.Spec.FeatureFlags = []wsmanapi.WorkspaceFeatureFlag{wsmanapi.WorkspaceFeatureFlag_PROTECTED_SECRETS}
```

Comment:
@Furisto should this be rebased? Since https://github.com/gitpod-io/gitpod/pull/14288 removed that feature flag.